### ISSUE TITLE
Name is not propagated to single Ingress

### ISSUE THREAD

**Issue by kadel** [kedgeproject/kedge#306]
If only one ingress is defined and doesn't have name the top name level should be used. 
This is correctly happening for Services, but not for Ingresses.

input:
```yaml
name: nginx
annotations:
  foo: deployment

containers:
- image: nginx

services:
- ports:
  - port: 8080
    targetPort: 80

ingresses:
- annotations:
    foo: ingress
  rules:
  - host: nginx.192.168.64.7.nip.io
    http:
      paths:
      - path: /
        backend:
          serviceName: nginx
          servicePort: 8080
```

output:
```
▶ ./kedge apply -f nginx.yaml
service "nginx" created
error: error when retrieving current configuration of:
&{0xc4200f6cc0 0xc4202ca5b0 default  STDIN 0xc4202fa638 0xc4202fa638  false}
from server for: "STDIN": resource name may not be empty
kubectl error: failed to execute command: exit status 1
```

generated ingress:
```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  creationTimestamp: null
  labels:
    app: nginx
spec:
  rules:
  - host: nginx.192.168.64.7.nip.io
    http:
      paths:
      - backend:
          serviceName: nginx
          servicePort: 8080
        path: /
status:
  loadBalancer: {}
```

**Comment by containscafeine** [comment]
Yeah, I realized that there is not `fixIngresses` function while doing ObjectMeta PR.

**Comment by containscafeine** [comment]
Closed in #310
